config.py

```diff
@@ -89,35 +128,40 @@
 OTHER_PLAYER_ITEM_POS_SIZE = 5920
 OTHER_PLAYER_SCALAR_SIZE = 8
 
+# Observation labels and dimensions
 OBSERVATION_LABELS = ["shop", "board", "bench", "states", "game_comp", "other_players"]
 POLICY_HEAD_SIZE = 2090
 NEEDS_2ND_DIM = [1, 2, 3, 4]
 
-MINIMUM_REWARD = float(environ.get("MINIMUM_REWARD", -300.0))
-MAXIMUM_REWARD = float(environ.get("MAXIMUM_REWARD", 300.0))
+# Reward boundaries
+MINIMUM_REWARD = get_float_env("MINIMUM_REWARD", -300.0)
+MAXIMUM_REWARD = get_float_env("MAXIMUM_REWARD", 300.0)
 
 class ModelConfig:
     # AI RELATED VALUES START HERE
-
     #### MODEL SET UP ####
-    HIDDEN_STATE_SIZE = int(environ.get("HIDDEN_STATE_SIZE", 1024))
-    NUM_RNN_CELLS = int(environ.get("NUM_RNN_CELLS", 2))
-    LSTM_SIZE = int(HIDDEN_STATE_SIZE / (NUM_RNN_CELLS * 2))
-    RNN_SIZES = [LSTM_SIZE] * NUM_RNN_CELLS
-    LAYER_HIDDEN_SIZE = int(environ.get("LAYER_HIDDEN_SIZE", 1024))
-    ROOT_DIRICHLET_ALPHA = float(environ.get("ROOT_DIRICHLET_ALPHA", 1.0))
-    ROOT_EXPLORATION_FRACTION = float(environ.get("ROOT_EXPLORATION_FRACTION", 0.25))
-    VISIT_TEMPERATURE = float(environ.get("VISIT_TEMPERATURE", 1.0))
-
-    PB_C_BASE = int(environ.get("PB_C_BASE", 19652))
-    PB_C_INIT = float(environ.get("PB_C_INIT", 1.25))
-
-    # ACTION_DIM = 10
-    ENCODER_NUM_STEPS = int(environ.get("ENCODER_NUM_STEPS", 601))
-    SELECTED_SAMPLES = eval(environ.get("SELECTED_SAMPLES", True))
-    MAX_GRAD_NORM = int(environ.get("MAX_GRAD_NORM", 5))
+    HIDDEN_STATE_SIZE = get_int_env("HIDDEN_STATE_SIZE", 1024)
+    NUM_RNN_CELLS = get_int_env("NUM_RNN_CELLS", 2)
+    # Placeholders for dynamic attributes. They will be set after the class definition.
+    LSTM_SIZE = None
+    RNN_SIZES = []
+
+    LAYER_HIDDEN_SIZE = get_int_env("LAYER_HIDDEN_SIZE", 1024)
+    ROOT_DIRICHLET_ALPHA = get_float_env("ROOT_DIRICHLET_ALPHA", 1.0)
+    ROOT_EXPLORATION_FRACTION = get_float_env("ROOT_EXPLORATION_FRACTION", 0.25)
+    VISIT_TEMPERATURE = get_float_env("VISIT_TEMPERATURE", 1.0)
+
+    PB_C_BASE = get_int_env("PB_C_BASE", 19652)
+    PB_C_INIT = get_float_env("PB_C_INIT", 1.25)
+
+    ENCODER_NUM_STEPS = get_int_env("ENCODER_NUM_STEPS", 601)
+    SELECTED_SAMPLES = get_bool_env("SELECTED_SAMPLES", "True")
+    MAX_GRAD_NORM = get_int_env("MAX_GRAD_NORM", 5)
 
     N_HEAD_HIDDEN_LAYERS = 2
+    NUM_SAMPLES = get_int_env("NUM_SAMPLES", 30)
+    NUM_SIMULATIONS = get_int_env("NUM_SIMULATIONS", 50)
 
-    NUM_SAMPLES = int(environ.get("NUM_SAMPLES", 30))
-    NUM_SIMULATIONS = int(environ.get("NUM_SIMULATIONS", 50))
+# Dynamically calculate LSTM_SIZE and RNN_SIZES based on current settings
+ModelConfig.LSTM_SIZE = int(ModelConfig.HIDDEN_STATE_SIZE / (ModelConfig.NUM_RNN_CELLS * 2))

```

Comment:
I think these need to be part of the model config. Basically just tabbed in once. If you create a new model config object for a new model. This should come as the default and None.

Same with RNN Sizes. It might already and I'm just not quite familiar enough with object default values but just double check for me.